File: Week_1/Reverse_String.java
import java.util.Arrays;
import java.util.Scanner;

public class Reverse_String {

    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length - 1;
        while (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.print("Enter a string: ");
            String inputString = scanner.nextLine();  

            char[] charArray = inputString.toCharArray();

            System.out.println("Original string: " + Arrays.toString(charArray));
            Reverse_String rs = new Reverse_String();
            rs.reverseString(charArray);
            System.out.println("Reversed string: " + Arrays.toString(charArray));  
        }
    }
}

File: Week_1/Reverse_string.py
from typing import List

class Solution:
    def reverseString(self, s: List[str]) -> None:
        left, right = 0, len(s) - 1
        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1
    def reverse(self, input_str: str) -> None:
        s = list(input_str)
        self.reverseString(s)
        reversed_string = ''.join(s)
        print(f"Original String: {input_str}")
        print(f"Reversed String: {reversed_string}")
input_string = input("Enter a string: ")
solution = Solution()
solution.reverse(input_string)
